src/routes/InstancesRoute.test.js

```diff
@@ -0,0 +1,315 @@
+import {
+  Layer,
+  Paneset
+} from '@folio/stripes/components';
+
+import renderWithIntl from '../../test/jest/helpers/renderWithIntl';
+import OverlayContainer from '../../test/helpers/OverlayContainer';
+import translationsProperties from '../../test/jest/helpers/translationsProperties';
+import { instances as instancesFixture } from '../../test/fixtures/instances';
+import { QUICK_EXPORT_LIMIT } from '../constants';
+import { DataContext } from '../contexts';
+import InstancesRoute from './InstancesRoute';
+
+const stripesStub = {
+  connect: Component => <Component />,
+  hasPerm: () => true,
+  logger: { log: noop },
+  locale: 'en-US',
+  plugins: {},
+};
+
+const InstancesRouteSetup = ({
+  instances = instancesFixture,
+  sendCallout = noop,
+  quickExportPOST = noop,
+} = {}) => (
+  <Router>
+    <StripesContext.Provider value={stripesStub}>
+      <CalloutContext.Provider value={{ sendCallout }}>
+        <DataContext.Provider value={{
+          contributorTypes: [],
+          instanceTypes: [],
+          locations: [],
+          instanceFormats: [],
+          modesOfIssuance: [],
+          natureOfContentTerms: [],
+          tagsRecords: [],
+        }}
+        >
+          <Paneset>
+            <Layer
+              isOpen
+              contentLabel="label"
+            >
+              <OverlayContainer />
+              <InstancesRoute
+                resources={{
+                  query: {
+                    query: '',
+                    sort: 'title',
+                  },
+                  records: {
+                    hasLoaded: true,
+                    resource: 'records',
+                    records: instances,
+                    other: { totalRecords: instances.length },
+                  },
+                  resultCount: instances.length,
+                  resultOffset: 0,
+                }}
+                mutator={{
+                  quickExport: { POST: quickExportPOST },
+                  resultCount: { replace: noop },
+                }}
+              />
+            </Layer>
+          </Paneset>
+        </DataContext.Provider>
+      </CalloutContext.Provider>
+    </StripesContext.Provider>
+  </Router>
+);
+
+describe('InstancesRoute', () => {
+  const quickExportAPICallMock = jest.fn(Promise.resolve.bind(Promise));
+  const sendCalloutMock = jest.fn();
+  let renderWithIntlResult;
+
+  describe('rendering InstancesRoute', () => {
+    beforeEach(() => {
+      renderWithIntlResult = renderWithIntl(
+        <InstancesRouteSetup
+          quickExportPOST={quickExportAPICallMock}
+          sendCallout={sendCalloutMock}
+        />,
+        translationsProperties
+      );
+    });
+
+    afterEach(() => {
+      jest.clearAllMocks();
+    });
+
+    it('should have proper list results size', () => {
+      expect(document.querySelectorAll('#pane-results-content .mclRowContainer > [role=row]').length).toEqual(3);
```

Comment:
Can we put fixture length instead of hardcoded value?

Reply:
Also, have you tried `getAllByRole('row')`?

Reply:
> Can we put fixture length instead of hardcoded value?

I put it on purpose. I think it makes sense to not bound to fixture data in tests. I did not do it for checking the content of the row and don't do it there for consistency.

Reply:
> Also, have you tried getAllByRole('row')?
Unfortunately, it did not work for me.